fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/UpdateExpression.java

```diff
@@ -83,12 +83,14 @@ public class UpdateExpression implements RelationalExpressionWithChildren, Plann
 
     @Nonnull
     private final Supplier<Integer> hashCodeWithoutChildrenSupplier;
+    private final boolean dryRun;
 
     public UpdateExpression(@Nonnull final Quantifier.ForEach inner,
                             @Nonnull final String targetRecordType,
                             @Nonnull final Type.Record targetType,
                             @Nonnull final Descriptors.Descriptor targetDescriptor,
-                            @Nonnull final Map<FieldValue.FieldPath, Value> transformMap) {
+                            @Nonnull final Map<FieldValue.FieldPath, Value> transformMap,
+                            final boolean dryRun) {
```

Comment:
I wouldn't model `dryRun` in plan operators since this is not relevant for planning. It is just a physical property that controls an execution workflow.

From planner's perspective, the update plan will (and should be) planned exactly the same way. I am thinking about adding this value to [`ExecuteProperties`](https://github.com/FoundationDB/fdb-record-layer/blob/main/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/ExecuteProperties.java) and set it when we're about to execute the physical plan.